Kernel/Memory/MemoryManager.h

```diff
@@ -307,7 +299,7 @@ class MemoryManager {
     PhysicalPageEntry* m_physical_page_entries { nullptr };
     size_t m_physical_page_entries_count { 0 };
 
-    RecursiveSpinlockProtected<GlobalData, LockRank::None> m_global_data;
+    SpinlockProtected<GlobalData, LockRank::None> m_global_data;
```

Comment:
This is locked twice during `find_rsdp_in_ia_pc_specific_memory_locations` if the RSDP wasn't found in the BIOS/EBDA. First by `MemoryManager::for_each_physical_memory_range` and then by `MemoryManager::allocate_mmio_kernel_region`.